docs/static_site/src/pages/get_started/osx_setup.md

```diff
@@ -24,176 +24,151 @@ permalink: /get_started/osx_setup
+xcode-select --install
+
+# Install Homebrew
+/usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
 
-Now we can start to build MXNet.
+# Install dependencies
+brew install cmake ninja ccache opencv
+```
 
-## Build the Shared Library
+`opencv` is an optional dependency. You can delete it from above `brew install`
+line and build MXNet without OpenCV support by setting `USE_OPENCV` to `OFF` in
+the configuration file described below.
 
-### Install MXNet dependencies
-Install the dependencies, required for MXNet, with the following commands:
-- [Homebrew](http://brew.sh/)
-- OpenBLAS and homebrew/core (for linear algebraic operations)
-- OpenCV (for computer vision operations)
 
-```bash
-	# Paste this command in Mac terminal to install Homebrew
-	/usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
+**Step 2:** Download MXNet sources and configure
 
-	# Insert the Homebrew directory at the top of your PATH environment variable
-	export PATH=/usr/local/bin:/usr/local/sbin:$PATH
-```
+Clone the repository:
 
 ```bash
-	brew update
-	brew install pkg-config
-	brew install graphviz
-	brew install openblas
-	brew tap homebrew/core
-	brew install opencv
-
-	# If building with MKLDNN
-	brew install llvm
-
-	# Get pip
-	easy_install pip
-	# For visualization of network graphs
-	pip install graphviz==0.8.4
-	# Jupyter notebook
-	pip install jupyter
+git clone --recursive https://github.com/apache/incubator-mxnet.git mxnet
+cd mxnet
+cp config/darwin.cmake config.cmake
 ```
 
-### Build MXNet Shared Library
+Please edit the config.cmake file based on your needs. The file contains a
+series of `set(name value CACHE TYPE "Description")` entries. For example, to
+build without Cuda, change `set(USE_CUDA ON CACHE TYPE "Build with CUDA
+support")` to `set(USE_CUDA OFF CACHE TYPE "Build with CUDA support")`.
+
+For a GPU-enabled build make sure you have installed the [CUDA dependencies
+first](#cuda-dependencies)). When building a GPU-enabled build on a machine
+without GPU, MXNet build can't autodetect your GPU architecture and will target
+all available GPU architectures. Please set the `MXNET_CUDA_ARCH` variable in
+`config.cmake` to your desired cuda architecture to speed up the build.
 
-After you have installed the dependencies, pull the MXNet source code from Git and build MXNet to produce an MXNet library called ```libmxnet.so```. You can clone the repository as described in the following code block, or you may try the [download links](download) for your desired MXNet version.
+To (optionally) build with MKL math library, please install MKL first based on
+the guide in [Math Library Selection](build_from_source#math-library-selection).
 
-The file called ```osx.mk``` has the configuration required for building MXNet on OS X. First copy ```make/osx.mk``` into ```config.mk```, which is used by the ```make``` command:
+**Step 3:** Build MXNet core shared library.
```

Comment:
These instructions and everything below seem to be identical to linux setup instruction. Could we consolidate them to reduce duplication?

Reply:
Agree. This will require more work though. Essentially the consolidated page will require client-side logic that allows users to select the operating system of their interest and based on the selection hides the unrelated instructions. Another option is file transclusion, but markdown may not support this.

Thus, the majority of the displayed consolidated page would be shared, and only the dependency installation would be OS specific.

Given this is not the focus of this PR, let's deduplicate in a separate PR once Windows instructions have been simplified as well.